### DISCUSSION TITLE
Some props not meant to be working?

### DISCUSSION THREAD

**Discussion by phungleson** [mantinedev/mantine#5110] [Q&A] [answered]
### What package has an issue

@mantine/core

### Describe the bug

Hi peeps,

I am playing around with some components and notice that some props don't seem to work like `variant` in `Checkbox` and `Radio`, i.e. changing the value does not change the UI.

From the code I can see it passed down to `InlineInput` then to another Box.. then I see `.input { appearance: none }` in the css so technically the variant is not meant to be working?

I am not sure if there is any other convention or future plan but if it is not meant to be working maybe it should be omitted from the props to improve DX?

Thanks,

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

7.1.5

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment, marked as answer]
`variant` prop is available on all components to allow adding custom variants – https://mantine.dev/styles/variants-sizes/
Some components have only one variant by default but allow you to add more.

**Comment by phungleson** [comment]
Ah ok thanks, so it also acts like a placeholder prop for further customisation.

**Comment by phungleson** [comment]
After thinking about this for a little while, wouldn't it be clearer if StyleAPI accepts both variant and data-variant?

variant is used for predefined set of customisation.

data-variant (which is any string) is used for user defined customisation.

So that in in case of Checkbox and Radio, they should accept data-variant, but should not accept variant. It would make better DX when developers don't have to wonder why variant does not work on Checkbox and Radio the way it should with other Inputs.

Users can override either variant or data-variant the same way with current implementation, i.e. using [data-variant=]

Maybe there are some cons that I haven't thought of besides some documentation to distinguish the two?
